packages/proxy/lib/http/util/cookies.ts

```diff
@@ -3,13 +3,49 @@ import type Debug from 'debug'
 import { URL } from 'url'
 import { cors } from '@packages/network'
 import { AutomationCookie, Cookie, CookieJar, toughCookieToAutomationCookie } from '@packages/server/lib/util/cookies'
+import { calculateSiteContext } from './top-simulation'
+import type { RequestCredentialLevel, RequestResourceType } from '../../types'
 
 interface RequestDetails {
   url: string
   isAUTFrame: boolean
   needsCrossOriginHandling: boolean
 }
 
+export const shouldAttachAndSetCookies = (requestUrl: string, AUTUrl: string | undefined, resourceType?: RequestResourceType, credentialLevel?: RequestCredentialLevel): boolean => {
+  if (!AUTUrl) return false
+
+  const siteContext = calculateSiteContext(requestUrl, AUTUrl)
+
```

Comment:
I don't have any here since the utility doesn't have a direct reference to `debug` (we could add one), but I do have quite a few in the middleware consuming the utility in #23888